coc/static/update_static.py

```diff
@@ -6,229 +6,1482 @@
+                elif is_super_troop: #for super troops use the original troop's lab level'
+                    original_troop = self.full_troop_data.get(super_troop_data["Original"])
+                    required_lab_level = original_troop.get(level).get("LaboratoryLevel")
+                    required_townhall = max_townhall_converter[required_lab_level]
+                elif is_seasonal_troop:
+                    required_lab_level = None
+                    required_townhall = level_data.get("UpgradeLevelByTH")
+                else:
+                    continue
+
+                new_level_data = {
+                    "level": int(level),
+                    "hitpoints": level_data.get("Hitpoints", 0),
+                    "dps": level_data.get("DPS", 0),
+
+                    "upgrade_time": upgrade_time_seconds,
+                    "upgrade_cost": level_data.get("UpgradeCost", 0),
+                    "required_lab_level": required_lab_level,
+                    "required_townhall": required_townhall,
+                }
+                hold_data["levels"].append(new_level_data)
+
+            if not hold_data["levels"]:
+                continue
+            new_troop_data.append(hold_data)
+
+        return new_troop_data
+
+    def _parse_guardian_data(self):
+        full_guardian_data = self.open_file("guardians.json")
+
+        new_guardian_data = []
+        for _id, (guardian_name, guardian_data) in enumerate(full_guardian_data.items(), 107000000):
+            if guardian_data.get("Deprecated", False):
+                continue
+            character_data = self.full_troop_data.get(guardian_data.get("CharacterDatas"))
+            hold_data = {
+                "_id": _id,
+                "name": self._translate(tid=guardian_data.get("TID")),
+                "info": self._translate(tid=guardian_data.get("InfoTID")),
+                "TID": {
+                    "name": guardian_data.get("TID"),
+                    "info": guardian_data.get("InfoTID"),
+                },
+                "upgrade_resource": self._parse_resource(resource=character_data.get("UpgradeResource")),
+
+                "is_flying": character_data.get("IsFlying"),
+                "is_air_targeting": character_data.get("AirTargets"),
+                "is_ground_targeting": character_data.get("GroundTargets"),
+
+                "movement_speed": character_data.get("Speed"),
+
+                "attack_speed": character_data.get("AttackSpeed"),
+                "attack_range": character_data.get("AttackRange"),
+                "levels": []
+            }
+
+            for level, level_data in character_data.items():
+                if not isinstance(level_data, dict):
+                    continue
+                #convert times to seconds, all times for all things will be in seconds
+                upgrade_time_seconds = level_data.get("UpgradeTimeH", 0) * 60 * 60
```

Comment:
lets just stick with the common code here and include days, minutes and seconds too. Just to make it future proof